R/names.R

```diff
@@ -0,0 +1,405 @@
+#' Retrieve and repair names
```

Comment:
Now that we've implemented this here, it doesn't feel quite right to me — name repair is most important for data frames, and this, the primary function, doesn't help at all. Maybe it would be better to call this `vec_rec_names()` or `vec_col_names()` or similar?

Reply:
Maybe we should just have `vec_as_names()` for the time being?

Reply:
I have unexported `vec_names()`.